### PR TITLE
PIP-70 Update to new Persephone template API

### PR DESCRIPTION
**Pull request by milt**
[PIP-70] Adapt to breaking changes introduced in [PERS-26] pr here: https://github.com/yetanalytics/project-persephone/pull/33

~~Also, persephone instrumentation sub-pr here: https://github.com/yetanalytics/project-persephone/pull/34~~ **MERGED**

- [x] Adapt template filter to new API
- [x] Resolve spec breakage (see instrumentation blacklist TODO)
- [x] Adapt concept filter to new API
- [x] Resolve new instrumentation breakage when running `validate-statement` directly from templates see https://github.com/yetanalytics/xapipe/runs/4588121289?check_suite_focus=true

[PIP-70]: https://yet.atlassian.net/browse/PIP-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PERS-26]: https://yet.atlassian.net/browse/PERS-26?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj [hunk 1]
```diff
@@ -57,57 +44,58 @@
                          ::contextCategoryActivityType
                          ::contextOtherActivityType
                          ::attachmentUsageType]))
-  :ret ::validator)
+  :ret ::t/template)
 
 (defn t-val
```
Comment (cliffcaseyyet) [inline src/lib/com/yetanalytics/xapipe/filter/concept.clj:49]: this name was supposed to mean template-validator. Maybe a rename to make-template or something?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
+    timeout-minutes: 5
```
Comment (cliffcaseyyet) [inline .github/workflows/ci.yml:16]: aggressive. what's its average? I guess this is PER job not for the whole combined thing so maybe i'm overestimating.

File: src/lib/com/yetanalytics/xapipe/filter.clj [hunk 3]
```diff
@@ -89,23 +91,20 @@
...
+                    (concat (concept/verb-templates vrb-ids)
+                            (concept/activity-type-templates act-ids)
+                            (concept/attachment-usage-templates att-ids))
+                    (some #{"Verb"} (set concept-types))
+                    (into (concept/verb-templates vrb-ids))
+                    (some #{"ActivityType"} (set concept-types))
+                    (into (concept/activity-type-templates act-ids))
+                    (some #{"AttachmentUsageType"} (set concept-types))
+                    (into (concept/attachment-usage-templates att-ids)))
+        validators (per/compile-templates->validators templates :validate-templates? false)]
```
Comment (cliffcaseyyet) [inline src/lib/com/yetanalytics/xapipe/filter.clj:105]: whats that key-arg do?

Comment (cliffcaseyyet) [reply src/lib/com/yetanalytics/xapipe/filter.clj:105]: I had a realization looking through pers. is this on account of the ID clash?

Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter.clj:105]: Yep! If you don't add it you get an (appropriate) error for non-unique template ids

Comment (cliffcaseyyet) [reply src/lib/com/yetanalytics/xapipe/filter.clj:105]: i could happily add some random to the id gen, if that's a nicer solution. I guess honestly validating a mock is kind of nonsense anyway so maybe not

Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter.clj:105]: Yeah I'd advise leaving it

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj (hunk 1 above)
Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj:49]: That's funny, I thought of it like `template-value` so I just left it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (milt) [reply .github/workflows/ci.yml:16]: Even when it has to load the whole NVD db from scratch it doesn't top 5 mins I believe (we'll see if this merges before that), but maybe we tune up if we find we need to

File: deps.edn [hunk 4]
```diff
@@ -13,9 +13,11 @@
                       org.clojure/data.json]}
         org.clojure/tools.logging {:mvn/version "1.1.0"}
         io.github.yetanalytics/project-persephone
-        {:git/sha "3a97133"
-         :git/tag "v0.7.3"
+        {:git/sha "387a11d44eec63b0ee7b2e25f6a2c002eed0fe21"
+         #_"3a97133"
+         #_#_:git/tag "v0.7.3"
```
Comment (kelvinqian00) [inline deps.edn:18]: Persephone v0.8.0 is out now, just so you know.

Comment (milt) [reply deps.edn:18]: Oh, I checked the list and didn't see it

Comment (kelvinqian00) [reply deps.edn:18]: I pushed the tag like 5 seconds ago

File: src/lib/com/yetanalytics/xapipe/filter.clj [hunk 5]
```diff
@@ -26,7 +26,9 @@
 (defn get-profile
   "Get a profile from the specified URL or throw"
   [url]
-  (try (per-json/json->edn (slurp url) :keywordize? true)
+  (try (json/parse-string-strict (slurp url) #(if (= % "@context")
+                                                :_context
+                                                (keyword nil %)))
```
Comment (kelvinqian00) [inline src/lib/com/yetanalytics/xapipe/filter.clj:31]: As I mentioned before Persephone now provides `coerce-profile` and `coerce-statement` in its `util.json` namespace.

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj [hunk 6]
```diff
@@ -57,57 +44,58 @@
...
-  "take a map of determining properties and returns a persephone validator with
+  "take a map of determining properties and returns a template with
   the determining properties merged onto a mock template"
   [det-props]
-  (p/template->validator (merge mock-template det-props)))
+  ;; TODO: This function is now private, update
+  (merge mock-template det-props))
```
Comment (kelvinqian00) [inline src/lib/com/yetanalytics/xapipe/filter/concept.clj:54]: It is now un-private (though it has been renamed to `compile-templates->validators` and now takes a coll).

Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj:54]: forgot to remove the todo, the call isn't even there anymore

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj (hunk 1 above)
Comment (kelvinqian00) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj:49]: Agree with Cliff. You see, unlike C, Clojure is not a Spartan language :)

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj (hunk 6 above)
Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj]: https://github.com/yetanalytics/xapipe/pull/68/commits/4adc694d35226074c8bc8795765ff56b651513ae

File: src/lib/com/yetanalytics/xapipe/filter.clj (hunk 5 above)
Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter.clj:31]: Yep, those are good to provide, but in this case it would be a little extra work. By providing awareness in the `key-fn` passed to cheshire we have the same effect

File: src/lib/com/yetanalytics/xapipe/filter/concept.clj (hunk 1 above)
Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj:49]: Sure, I'll defer to the original author. `make-template` it is

Comment (milt) [reply src/lib/com/yetanalytics/xapipe/filter/concept.clj]: https://github.com/yetanalytics/xapipe/pull/68/commits/a3af45434926957c3ebe80fc4e0052e7fdec5af6

File: dev-resources/profiles/calibration_concept.jsonld [hunk 7]
```diff
@@ -159,10 +159,10 @@
       "type" : "StatementTemplate",
       "inScheme" : "https://xapinet.org/xapi/yet/calibration-concept/v1",
       "prefLabel" : {
-        "en" : "Object Activity Type Template 2"
+        "en-US" : "Object Activity Type Template 2"
```
Comment (kelvinqian00) [inline dev-resources/profiles/calibration_concept.jsonld:162]: @cliffcaseyyet Curious why the change from `en` to `en-US`?

Comment (milt) [reply dev-resources/profiles/calibration_concept.jsonld:162]: MURICA 🇺🇸 🎆  thats why

**Review by kelvinqian00 — APPROVED** [review]
Looks good to me.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (milt) [reply .github/workflows/ci.yml]: OK, it's now set to 10 mins

**Review by cliffcaseyyet — APPROVED** [review]
